presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -267,4 +278,37 @@ private Expression coerceIfNecessary(Expression original, Expression rewritten)
             }
         }, expression);
     }
+
+    public FunctionCall rewriteGroupingOperationToFunctionCall(GroupingOperation expression, QuerySpecification node)
```

Comment:
`TranslationMap` doesn't feel like a proper place for `rewriteGroupingOperationToFunctionCall`. It probably should be part of `GroupingOperationRewriter`. Why do you need it to be here?